src/ucs/config/parser.c

```diff
@@ -1512,16 +1525,63 @@ static char *ucs_config_get_value_from_config_file(const char *name)
     return kh_val(&ucs_config_file_vars, iter);
 }
 
+static int ucs_config_parse_check_filter(const char *name, const char *value)
+{
+    typedef struct {
+        char *name;
+        const char *(*value_f)();
+    } filter_t;
+
+    static filter_t filters[] = {{UCS_CPU_VENDOR_LABEL, ucs_cpu_vendor_name},
+                                 {UCS_CPU_MODEL_LABEL, ucs_cpu_model_name},
+                                 {UCS_SYS_DMI_PRODUCT_NAME_LABEL,
+                                  ucs_sys_dmi_product_name}};
+
+    filter_t *filter;
+
+    ucs_carray_for_each(filter, filters, ucs_static_array_size(filters)) {
+        if ((strcmp(name, filter->name) == 0) &&
+            (fnmatch(value, filter->value_f(), FNM_CASEFOLD) != 0)) {
+            return 1;
+        }
+    }
+
+    return 0;
+}
+
+static int
+ucs_config_parse_skip_section(ucs_config_parse_section_t *section_info,
+                              const char *section, const char *name,
+                              const char *value)
+{
+    if (strcmp(section, section_info->name) != 0) {
+        ucs_strncpy_zero(section_info->name, section,
+                         sizeof(section_info->name));
+        section_info->skip = 0;
+    } else if (section_info->skip) {
+        return 1;
+    }
+
+    section_info->skip = ucs_config_parse_check_filter(name, value);
+    return section_info->skip;
+}
+
 static int ucs_config_parse_config_file_line(void *arg, const char *section,
                                              const char *name,
                                              const char *value)
 {
-    khiter_t iter = kh_get(ucs_config_map, &ucs_config_file_vars, name);
-    int override  = *(int*)arg;
+    ucs_config_parse_arg_t *parse_arg = (ucs_config_parse_arg_t*)arg;
+    khiter_t iter;
     int result;
 
+    if (ucs_config_parse_skip_section(&parse_arg->section_info, section, name,
```

Comment:
optional: IMO the code could be a little simpler if ucs_config_parse_skip_section returned void, and we checked parse_arg->section_info.skip directly in ucs_config_parse_config_file_line.

Reply:
Updated: `int ucs_config_parse_skip_section` -> `void ucs_config_parse_set_section_info`.